experimental/single_number_formatter/src/ule.rs

```diff
@@ -0,0 +1,130 @@
+pub struct CurrencyPatternsULE([u8; 3]);
+
+impl CurrencyPatternsULE {}
+
+// Safety (based on the safety checklist on the ULE trait):
+//  1. CurrencyPatternsULE does not include any uninitialized or padding bytes.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  2. CurrencyPatternsULE is aligned to 1 byte.
+//     (achieved by `#[repr(transparent)]` on a ULE type)
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid.
+//  4. The impl of validate_byte_slice() returns an error if there are extra bytes.
+//  5. The other ULE methods use the default impl.
+//  6. CurrencyPatternsULE byte equality is semantic equality.
+unsafe impl ULE for CurrencyPatternsULE {
+    fn validate_byte_slice(bytes: &[u8]) -> Result<(), ZeroVecError> {
+        if bytes.len() % 3 != 0 {
+            return Err(ZeroVecError::length::<Self>(bytes.len()));
+        }
+
+        Ok(())
+    }
+}
+
+const PATTERN_SHORT_POSITION: u8 = 6;
+const PATTERN_NARROW_POSITION: u8 = 4;
+const INDEX_SHORT_POSITION: u8 = 2;
+const INDEX_NARROW_POSITION: u8 = 0;
+
+impl AsULE for CurrencyPatterns {
+    type ULE = CurrencyPatternsULE;
+
+    #[inline]
+    fn to_unaligned(self) -> Self::ULE {
+        let mut first_byte_ule: u8 = 0;
+
+        if self.short_pattern_standard == PatternSelection::StandardAlphaNextToNumber {
+            first_byte_ule |= 0b01 << PATTERN_SHORT_POSITION;
+        }
+        if self.narrow_pattern_standard == PatternSelection::StandardAlphaNextToNumber {
+            first_byte_ule |= 0b01 << PATTERN_NARROW_POSITION;
+        }
+
+        // For short_place_holder_index
+        let [first_index_byte, second_byte_ule] = self.short_place_holder_index.to_be_bytes();
+        if first_index_byte < 0b0100 {
+            first_byte_ule |= first_index_byte << INDEX_SHORT_POSITION;
+        } else {
+            panic!(
+                "short_place_holder_index is too large {}, {}",
+                self.short_place_holder_index, first_index_byte
+            )
+        }
+
+        // For narrow_place_holder_index
+        let [first_index_byte, third_byte_ule] = self.narrow_place_holder_index.to_be_bytes();
+        if first_index_byte < 0b0100 {
+            first_byte_ule |= first_index_byte << INDEX_NARROW_POSITION;
+        } else {
+            panic!(
+                "narrow_place_holder_index is too large {}, {}",
+                self.narrow_place_holder_index, first_index_byte
+            )
+        }
+        CurrencyPatternsULE([first_byte_ule, second_byte_ule, third_byte_ule])
+    }
+
+    #[inline]
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        let [first_byte, second_byte, third_byte] = unaligned.0;
+
+        let short_pattern_standard =
+            if (first_byte & 0b11 << PATTERN_SHORT_POSITION) >> PATTERN_SHORT_POSITION == 1 {
+                PatternSelection::StandardAlphaNextToNumber
+            } else {
+                PatternSelection::Standard
+            };
```

Comment:
Nit (optional): I would find this more readable as something similar to what you had before:

```rust
if (first_byte & 0b1100_0000) == 0b0100_0000 {
    PatternSelection::StandardAlphaNextToNumber
} else {
    PatternSelection::Standard
};
```
and similar below. However, it is nice how you got this to be parallel with `INDEX_SHORT_POSITION` and `INDEX_NARROW_POSITION` so feel free to ignore this suggestion.

Reply:
I have changed to 
```
        let short_pattern_standard =
            if first_byte & (0b11 << PATTERN_SHORT_SHIFT) == 0b01 << PATTERN_SHORT_SHIFT {
                PatternSelection::StandardAlphaNextToNumber
            } else {
                PatternSelection::Standard
            };
```

which more readable.